src/pybind/mgr/mgr_module.py

```diff
@@ -610,8 +612,21 @@ def __del__(self):
         unconfigure_logger()
 
     @classmethod
-    def _register_commands(cls):
+    def _register_commands(cls, module_name):
+        cls.MODULE_OPTIONS.append(Option(name='log_level', type='str', default="WARNING", runtime=True))
+
         cls.COMMANDS.extend(CLICommand.dump_cmd_list())
+        cls.COMMANDS.extend([{
+            'cmd': '{} set log_level '
+                   'name=level,type=CephChoices,'
+                   'strings=INFO|DEBUG|CRITICAL|ERROR|WARNING|NOTSET'.format(module_name),
+            'desc': 'Set the module python log level',
+            'perm': 'w'
+        }, {
+            'cmd': '{} get log_level'.format(module_name),
+            'desc': 'Show the module python log level',
+            'perm': 'r'
+        }])
```

Comment:
Instead of doing special commands, can we instead do
```
ceph config set mgr mgr/$module/log_level debug
```
?  That is a simpler change to each module (just add an item to MODULE_OPTIONS) and keeps things in the config realm without polluting the command namespace

Reply:
@epuertat Yes!  ``config_notify()`` gets called when an option changes.  see mgr/hello/module.py for an example.  The pattern is usually to call it from ``__init__`` and make the implemention make setting ``foo`` copied to ``self.foo`` for maximum ease of use.

Reply:
I was not aware of `config_notify`, in that case I can drop the commands.

Reply:
@liewegas thanks for tip! That's exactly what I was looking for.

@rjfd regarding the concern on using `get_module_option` vs. a proxy variable, I just read this in the [ceph-mgr docs](https://docs.ceph.com/docs/master/mgr/modules/#configuration-options):

> Reads are fast: ceph-mgr keeps a local in-memory copy, so in many cases you can just do a get_module_option every time you use a option, rather than copying it out into a variable.

We should (at some time) run a benchmark on these assumptions to ensure the API overhead is that thin, but if that statement still applies (I assume those are @jcsp's words) we could simplify some things we are doing in dashboard.